EPR/Library/tests/Makefile

```diff
@@ -2,29 +2,36 @@
 # this is the EPR/Library/tests Makefile
 #https://stackoverflow.com/a/28236905
 #https://gcc.gnu.org/onlinedocs/gcc/Invoking-G_002b_002b.html
+
 #for debugging information, add -g to this next line or in a recipe, below
-CC = g++ -std=c++17 -Wall -Wextra -Werror -O2 -DSGI32
+CC = g++ -std=c++17 -Wall -Wextra -O2 -DSGI32
 LIB = ../../Library
 INCQ = -iquote ../
 
-libtest:
-	$(CC) -g $(INCQ) libtest.cpp ../err_msg.cpp ../octree.cpp -o test
-
-test_atomics: $(LIB)/atomics.hpp test_atomics.cpp
-	$(CC) $(INCQ) -L$(LIB) \
-	test_atomics.cpp -o test_atomics -lm
 
-test_rays: 
+test_atomics:
 	$(CC) $(INCQ) -L$(LIB) \
-	test_rays.cpp ../rays.cpp -o test_rays -lm
+	$(LIB)/atomics.cpp test_atomics.cpp -o test_atomics -lm
 
 test_noise: 
 	$(CC) -g $(INCQ) -L$(LIB) \
-	test_noise.cpp ../noise.cpp -o test_noise -lm
+	test_noise.cpp $(LIB)/noise.cpp -o test_noise -lm
 
+test_octree: #TODO change libtest.cpp to test_octree.cpp
+	$(CC) -g $(INCQ) -L$(LIB) \
+  $(LIB)/err_msg.cpp $(LIB)/octree.cpp libtest.cpp -o test_octree
+  
 test_raster:
 	$(CC) $(INCQ) -L$(LIB) \
-	test_raster.cpp ../raster.cpp -o test_raster -lm
+	test_raster.cpp $(LIB)/raster.cpp -o test_raster -lm
 
+test_rays: 
+	$(CC) $(INCQ) -L$(LIB) \
+	test_rays.cpp $(LIB)/rays.cpp -o test_rays -lm
+  
+ test_stack:
+	$(CC) -g $(INCQ) -L$(LIB) \
+	$(LIB)/octree.c $(LIB)/stack.cpp test_stack.cpp -o test_stack -lm
+  
 clean:
-	rm -f test_atomics test_rays test_noise test test_raster
+	find . -maxdepth 1 -name "test_*" -not -name "*\.cpp" -print -delete
```

Comment:
I think that 'find' as written will not remove the output of libtest, which recipe says '-o test'.
Should we change the libtest recipe to output something that matches the regex 'test_*'?
Is the '-not' syntax portable, or is that a recent GNU extension?
Should we instead use '!' as POSIX compliant?
Same question regarding '-delete'. Wouldn't '-execdir rm -f {} \;' be portable?

Reply:
I think that you have a good point with POSIX compliance - if I'm being honest, I looked up a couple regex solutions, and I definitely don't know enough to be able to produce something I'm confident in. Given that Genetti will be the main user of this code, I feel confident in his ability to delete files if this doesn't work (although I will be sad, because I like this solution).

Also, I totally agree with your point about libtest, I think that we should change that to test_octree.

Reply:
Updated to use rm -f